couchdb/source/pkg/adapter/adapter.go

```diff
@@ -91,47 +113,67 @@ func newAdapter(ctx context.Context, env *envConfig, ceClient cloudevents.Client
 
 		couchDB: db,
 		source:  env.EventSource,
+		feed:    env.Feed,
 		options: map[string]interface{}{
-			"feed":  "normal",
+			"feed":  env.Feed,
 			"since": "0",
 		},
 	}
 }
 
 func (a *couchDbAdapter) Start(stopCh <-chan struct{}) error {
-	// Just Poll for now. Other modes (long running, continues) will be provided later on.
-	return wait.PollUntil(2*time.Second, a.send, stopCh)
+	period := 2 * time.Second
+	if a.feed == "continuous" {
+		a.options["heartbeat"] = 6000
+		period = 0
+	}
+	wait.Until(a.processChanges, period, stopCh)
+	return nil
 }
 
-func (a *couchDbAdapter) send() (done bool, err error) {
+func (a *couchDbAdapter) processChanges() {
 	changes, err := a.couchDB.Changes(context.TODO(), a.options)
 	if err != nil {
-		return false, err
+		a.logger.Error("Error getting the list of changes", zap.Error(err))
+		return
 	}
 
 	for changes.Next() {
-		event, err := a.makeEvent(changes)
-		if err != nil {
-			return false, err
-		}
+		if changes.Seq() != "" {
+			event, err := a.makeEvent(changes)
 
-		if _, _, err := a.ce.Send(context.Background(), *event); err != nil {
-			a.logger.Info("event delivery failed", zap.Error(err))
-			return false, err
-		}
+			if err != nil {
+				a.logger.Error("error making event", zap.Error(err))
+			}
 
-		a.options["since"] = changes.Seq()
+			if _, _, err := a.ce.Send(context.TODO(), *event); err != nil {
+				a.logger.Error("event delivery failed", zap.Error(err))
+			}
+
+			a.options["since"] = changes.Seq()
+		}
 	}
 
-	return false, nil
+	if changes.Err() != nil {
+		if changes.Err() == io.EOF {
+			a.logger.Error("The connection to the changes feed was interrupted.", zap.Error(changes.Err()))
+		} else {
+			a.logger.Error("Error found in the changes feed.", zap.Error(changes.Err()))
+		}
+	}
 }
 
 func (a *couchDbAdapter) makeEvent(changes *kivik.Changes) (*cloudevents.Event, error) {
 	event := cloudevents.NewEvent(cloudevents.VersionV03)
 	event.SetID(changes.Seq())
 	event.SetSource(a.source)
```

Comment:
not _directly_ related to _THIS_ RP,

but: relevant:   https://github.com/knative/eventing-contrib/pull/663/files#r337488954

Reply:
I'm happy to change when the Prometheus source is merged.